runtime/src/accounts_db.rs

```diff
@@ -1221,14 +1226,14 @@ impl AccountsDB {
     {
         let mut collector = A::default();
         let accounts_index = self.accounts_index.read().unwrap();
-        let storage = self.storage.read().unwrap();
         accounts_index.scan_accounts(ancestors, |pubkey, (account_info, slot)| {
-            scan_func(
-                &mut collector,
-                storage
-                    .scan_accounts(account_info, slot)
-                    .map(|(account, slot)| (pubkey, account, slot)),
-            )
+            let account_storage_entry = self
+                .storage
+                .get_account_storage_entry(slot, account_info.store_id);
+            let account_slot = account_storage_entry
+                .and_then(|account_storage_entry| account_storage_entry.get_account(account_info))
+                .map(|account| (pubkey, account, slot));
```

Comment:
nits: how about defining `AccountsDb::get_account_from_storage()` with these code and share with [that](https://github.com/solana-labs/solana/pull/12126/files#r503739183)?

Reply:
cleaned up!